File: app/src/main/java/com/ssbprep/activities/InterviewActivity.java
package com.ssbprep.activities;

import android.os.Bundle;
import android.widget.Button;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.ssbprep.R;

public class InterviewActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_interview);

        // Initialize all interview blocks
        Button introButton = findViewById(R.id.introButton);
        Button tipsButton = findViewById(R.id.tipsButton);
        Button questionsButton = findViewById(R.id.questionsButton);
        Button ciqButton = findViewById(R.id.ciqButton);
        Button backButton = findViewById(R.id.backButton);

        // Set click listeners
        introButton.setOnClickListener(v -> handleBlockClick("Introduction"));
        tipsButton.setOnClickListener(v -> handleBlockClick("Tips and Techniques"));
        questionsButton.setOnClickListener(v -> handleBlockClick("Questions Asked"));
        ciqButton.setOnClickListener(v -> handleBlockClick("CIQ's (Combined Interview Questionnaire)"));
        backButton.setOnClickListener(v -> onBackPressed());
    }

    private void handleBlockClick(String blockName) {
        Toast.makeText(this, blockName + " selected!", Toast.LENGTH_SHORT).show();
        // TODO: Navigate to block details screen
    }
}

File: app/src/main/java/com/ssbprep/activities/InterviewActivity.kt
package com.ssbprep.activities

import android.content.Intent
import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import com.ssbprep.databinding.ActivityInterviewBinding
import com.ssbprep.viewmodel.PaymentViewModel

class InterviewActivity : AppCompatActivity() {

    private lateinit var binding: ActivityInterviewBinding
    private val viewModel: PaymentViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityInterviewBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.backButton.setOnClickListener { finish() }

        binding.scheduleInterviewButton.setOnClickListener {
            checkPaymentAndNavigate()
        }

        // Module Listeners
        binding.prepGuideButton.setOnClickListener {
            val intent = Intent(this, ServiceDetailActivity::class.java)
            intent.putExtra("SERVICE_TYPE", "INTERVIEW_GUIDE")
            startActivity(intent)
        }
        
        binding.ciqButton.setOnClickListener {
            startActivity(Intent(this, CIQsActivity::class.java))
        }

        binding.swotButton.setOnClickListener {
            val intent = Intent(this, ServiceDetailActivity::class.java)
            intent.putExtra("SERVICE_TYPE", "SWOT")
            startActivity(intent)
        }
    }

    private fun checkPaymentAndNavigate() {
        if (viewModel.isPaid.value) {
            startActivity(Intent(this@InterviewActivity, InterviewBookingActivity::class.java))
        } else {
            val intent = Intent(this, PaymentActivity::class.java)
            intent.putExtra("USER_ID", viewModel.getUserId())
            startActivity(intent)
        }
    }
}
